alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessorV2.java

```diff
@@ -0,0 +1,225 @@
+        }
+
+        List<DistributionJobModel> configurationJobModels = pageOfJobsWithDescriptorNames.map(this::convertToDistributionJobModel).getContent();
+        return new AlertPagedModel<>(pageOfJobsWithDescriptorNames.getTotalPages(), pageNumber, pageLimit, configurationJobModels);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<DistributionJobModel> getJobById(UUID jobId) {
+        return distributionJobRepository.findById(jobId).map(this::convertToDistributionJobModel);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<DistributionJobModel> getJobByName(String jobName) {
+        return distributionJobRepository.findByName(jobName)
+                   .map(this::convertToDistributionJobModel);
+    }
+
+    @Override
+    public List<DistributionJobModel> getMatchingEnabledJobs(FrequencyType frequency, Long providerConfigId, NotificationType notificationType) {
+        // TODO change this to return a page of jobs
+        return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(frequency.name(), providerConfigId, notificationType.name()));
+    }
+
+    @Override
+    public List<DistributionJobModel> getMatchingEnabledJobs(Long providerConfigId, NotificationType notificationType) {
+        // TODO change this to return a page of jobs
+        return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(providerConfigId, notificationType.name()));
+    }
+
+    private List<DistributionJobModel> getMatchingEnabledJobs(Supplier<List<DistributionJobEntity>> getJobs) {
+        // TODO change this to return a page of jobs
+        List<DistributionJobEntity> matchingEnabledJob = getJobs.get();
+        return matchingEnabledJob
+                   .stream()
+                   .map(this::convertToDistributionJobModel)
+                   .collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public DistributionJobModel createJob(Collection<String> descriptorNames, Collection<ConfigurationFieldModel> configuredFields) {
+        return createJobWithId(null, configuredFields, DateUtils.createCurrentDateTimestamp(), null);
+    }
+
+    @Override
+    @Transactional
+    public DistributionJobModel updateJob(UUID jobId, Collection<String> descriptorNames, Collection<ConfigurationFieldModel> configuredFields) throws AlertDatabaseConstraintException {
+        DistributionJobEntity jobEntity = distributionJobRepository.findById(jobId)
+                                              .orElseThrow(() -> new AlertDatabaseConstraintException(String.format("No job exists with the id [%s]", jobId.toString())));
+        OffsetDateTime createdAt = jobEntity.getCreatedAt();
+
+        deleteJob(jobId);
+        return createJobWithId(jobId, configuredFields, createdAt, DateUtils.createCurrentDateTimestamp());
+    }
```

Comment:
I think we want to take in a different model rather than the "old" way of `Collection<String> descriptorNames` and `Collection<ConfigurationFieldModel> configuredFields`. Something along the lines of `DistributionJobRequestModel` that has a subset of the fields of `DistributionJobModel`. This model would not have `jobId`, `createdAt` or `lastUpdated` fields.